src/com/facebook/buck/ide/intellij/IjProjectCommandHelper.java

```diff
@@ -233,6 +252,127 @@ private TargetGraph getProjectGraphForIde(
+                .collect(ImmutableSet.toImmutableSet()));
+    if (exitCode != ExitCode.SUCCESS) {
+      return exitCode;
+    }
+
+    copyGeneratedGoCode(targetGraphAndTargets, generatedPackages);
+    return ExitCode.SUCCESS;
+  }
+
+  /**
+   * Assuming GOPATH is set to a directory higher or equal to buck root, copy generated code to the
+   * package path relative to the highest level vendor directory. Not handling the case of GOPATH
+   * lower than buck root for now, as it requires walking the directory structure, which can be
+   * expensive and unreliable (e.g., what if there are multiple src directory?).
+   */
+  private void copyGeneratedGoCode(
+      TargetGraphAndTargets targetGraphAndTargets,
+      Map<BuildTargetSourcePath, Path> generatedPackages)
+      throws IOException {
+    Path vendorPath;
+    ProjectFilesystem fs = cell.getFilesystem();
+    if (fs.exists(Paths.get("src"))) {
+      vendorPath = Paths.get("src", "vendor");
+    } else {
+      vendorPath = Paths.get("vendor");
+    }
+    ActionGraphAndBuilder result =
+        Preconditions.checkNotNull(getActionGraph(targetGraphAndTargets.getTargetGraph()));
+    DefaultSourcePathResolver sourcePathResolver =
+        DefaultSourcePathResolver.from(new SourcePathRuleFinder(result.getActionGraphBuilder()));
+    for (BuildTargetSourcePath sourcePath : generatedPackages.keySet()) {
+      Path desiredPath = vendorPath.resolve(generatedPackages.get(sourcePath));
+      fs.mkdirs(desiredPath);
+      for (Path f : fs.getDirectoryContents(desiredPath)) {
+        if (fs.isFile(f)) {
+          fs.deleteFileAtPath(f);
+        }
+      }
+      Path generatedSrc = sourcePathResolver.getAbsolutePath(sourcePath);
+      if (fs.isDirectory(generatedSrc)) {
+        fs.copyFolder(generatedSrc, desiredPath);
+      } else {
+        fs.copyFile(generatedSrc, desiredPath.resolve(generatedSrc.getFileName()));
+      }
+    }
+  }
+
+  /**
+   * Find code generation targets by inspecting go_library targets in the target graph with "srcs"
+   * pointing to other Buck targets rather than Go files. Those Buck targets in "srcs" are assumed
+   * to be code generation targets. Their output is intended to be used as some package name, either
+   * specified by package_name argument of go_library, or guessed from the base path of the the
+   * go_library
+   */
+  @SuppressWarnings("unchecked")
+  private Map<BuildTargetSourcePath, Path> findCodeGenerationTargets(
+      TargetGraphAndTargets targetGraphAndTargets) {
+    ClassLoader classLoader = pluginManager.getPluginClassLoader("com.facebook.buck.features.go");
+    Map<BuildTargetSourcePath, Path> generatedPackages = new HashMap<>();
+    try {
+      Class<?> goLibraryDescriptionArgClass;
+      if (classLoader != null) {
+        goLibraryDescriptionArgClass =
+            Class.forName(
+                "com.facebook.buck.features.go.GoLibraryDescriptionArg", true, classLoader);
```

Comment:
Could you wait with that? I'll try to migrate project command to a plugin model, but it can take some time.

Reply:
Do you see any difficulties with changing this in future?

Reply:
We really need this feature now. Since changes in this PR is very localized, it is easy to change whenever you migrate project command. I am happy to assist you for migrating this piece of code.